### DISCUSSION TITLE
How do I sort labels?

### DISCUSSION THREAD

**Discussion by slalaurette** [martinrotter/rssguard#1013] [Q&A]
It appears that RSS Guard sorts labels automatically, which doesn't let me organize them in a convenient way. I need to add dozens of labels in a very specific fashion so I would need for them to be sortable. 

After adding the first few labels, I decided to go with the (ugly, but usually effective) workaround of prepending numbers to the labels so that they would become ordered. To my dismay, this didn't work. It's like RSS Guard completely ignores numbers and sorts labels only by their letters.

![clip_20230717_16 48 18](https://github.com/martinrotter/rssguard/assets/12722382/5e89fd1d-d96e-49c8-97d4-b51ec93208f8)

(Or, actually, seeing this screenshot, the ordering is not quite alphabetical either, now that I notice.)

This is unexpected and very frustrating, because it would seem that there is actually no way to have my labels sorted as I want them. Is this so? Is there any workaround for this?

**Comment by slalaurette** [comment]
It appears that I cannot delete labels either. Is this correct?

**Comment by martinrotter** [comment]
At this point labels should be sorted alphabetically and they can be removed but only in some "account" types.

For example classic RSS/ATOM/JSON account of course can add/edit/remove labels, other account types at this point cannot (because they use remote labels), but those individual operations could be added.

**Comment by slalaurette** [comment]
Well, I am using a classic RSS/ATOM/JSON account and what I noticed right now is that I can remove labels, but I couldn't in the previous session. Alphabetical sorting is also working now. 

Somehow it looks like RSS Guard needs to be restarted before I can remove labels and before the automatic sorting takes place.

I had noticed a similar thing weeks ago, where the option for editing an item was disabled until I restarted the program. I don't remember the details right now, but it was when I tried to edit a feed's properties.

**Reply by martinrotter** [reply]
Can you perhaps provide video of labels not being sorted? It would be great to have exact steps to reproduce problem with deleting too so that I can reproduce.

**Comment by slalaurette** [comment]
Well, now I switched windows to another program, and when I came back to RSS Guard the options for editing and removing labels were disabled again. They aren't auto-sorting anymore, either.
